### PR TITLE
Improve log messages for data directory and configuration details

### PR DESCRIPTION
**Pull request by anipaul2**
Following up from https://github.com/talaia-labs/rust-teos/pull/210

fixes: https://github.com/talaia-labs/rust-teos/issues/202

### CODE REVIEW

**Comment by anipaul2** [comment]
> Following up from #210
> 
> fixes: #202

**@mariocynicys @sr-gi** Apologies for the confusion in my recent pull requests. I've been having difficulties with Git rebase, which has caused some issues in my submissions. I'm currently working on understanding the process better. Please consider accepting this PR as I am committed to learning from this experience and ensuring that future contributions are free of similar issues. Your support and understanding in this matter are greatly appreciated, and I am confident that my future work will reflect the improvements I am making. Thank you for your patience and guidance.

**Comment by mariocynicys** [comment]
Please fix the lint issues.

**Comment by anipaul2** [comment]
> Please fix the lint issues.

Done

File: teos/src/config.rs [hunk 1]
```diff
@@ -242,6 +242,27 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// Logs non-default options.
+    pub fn log_non_default_options(&self) {
+        let json_default_config = serde_json::json!(&Config::default());
+        let json_config = serde_json::json!(&self);
+        let protected_keys = ["btc_rpc_user", "btc_rpc_password"];
```
Comment (sr-gi) [inline teos/src/config.rs:250]: nit: Let's call this `sensitive_args` now that we know that's how core calls them.

**Review by sr-gi — CHANGES_REQUESTED** [review]
Just a tinny nit, LGTM otherwise.

**Review by mariocynicys — APPROVED** [review]

File: teos/src/config.rs (hunk 1 above)
Comment (anipaul2) [reply teos/src/config.rs:250]: > nit: Let's call this `sensitive_args` now that we know that's how core calls them.

**protected_keys** has to be changed to **sensitive_args**?

Comment (sr-gi) [reply teos/src/config.rs:250]: Yes

Comment (anipaul2) [reply teos/src/config.rs]: > Yes

done

**Review by mariocynicys — APPROVED** [review]

**Review by sr-gi — APPROVED** [review]

**Comment by sr-gi** [comment]
@anipaul2 commits need to be signed in order to be merged

**Comment by anipaul2** [comment]
> @anipaul2 commits need to be signed in order to be merged

Done.

**Comment by anipaul2** [comment]
Thank you @sr-gi @mariocynicys for the continuous guidance, and I'm delighted that the PR has been finally merged. I am committed to learning from my mistakes and look forward to making further contributions to the project.

**Pull request merged**